src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/StartupObjectsEnumProvider.cs

```diff
@@ -31,7 +30,18 @@ public StartupObjectsEnumProvider([Import(typeof(VisualStudioWorkspace))] Worksp
 
         public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<NameValuePair>? options)
         {
-            return Task.FromResult<IDynamicEnumValuesGenerator>(new StartupObjectsEnumGenerator(_workspace, _unconfiguredProject));
+            // We only include a value representing the "not set" state if requested. This is
+            // because the old property pages explicitly add the "(Not set)" value at the UI
+            // layer; the new property pages do not have that option and so the value must come
+            // from the enum provider.
+            // When this project system no longer needs to support the old property pages we can
+            // remove this and always include the "(Not set)" value.
+            bool includeEmptyValue = options?.Any(pair =>
+                pair.Name == "IncludeEmptyValue"
+                && bool.TryParse(pair.Value, out bool optionValue)
+                && optionValue) ?? false;
```

Comment:
Why not update the legacy pages to not add this item?

Reply:
The legacy property pages need to work with both the legacy project system _and_ dotnet/project-system. If this item isn't added at the UI level, then I would need to update the legacy project system to add it, just like I've done here. I don't want to touch either if I can avoid it, especially when most of the complexity I've added here (around whether or not to add the "(Not set)" item) can be removed in the Dev17 time frame.